### PR TITLE
Removes old images on remote host, closes #24

### LINKED ISSUE #24: Clean up old unused images on the server
At the moment we just push images to the server, but we never clean them up.

I just ran

    docker system prune -a --volumes

and cleaned up roughly 3 GB of images.

We should look into how to clean up old unused images.

Maybe we then can have a cron job on the server (or Travis?) clean up for us on a regular basis.

### PR DESCRIPTION
**Pull request by daniel-j-h**
For #24. This deletes all images on the server before loading the new one.

If we want to delete specific images (e.g. based on time, etc.) we'd have to use the filter expression or think about something more complicated. I think for now this should be Good Enough :tm:

### CODE REVIEW

**Comment by daniel-j-h** [comment]
@ucyo maybe you have an idea how to better do this, the problem is
- without the `-a` flag we remove images which are not tagged and images not currently in use
- but our app images are always tagged staging and production

File: .travis.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ after_success:
       ssh-keyscan -t ed25519 "${APP_DEPLOY_HOST}" >> ~/.ssh/known_hosts
 
       docker tag usedgoodreads/app usedgoodreads/app:staging
-      docker image save usedgoodreads/app:staging | ssh "${APP_DEPLOY_USER}@${APP_DEPLOY_HOST}" "docker image load"
+      docker image save usedgoodreads/app:staging | ssh "${APP_DEPLOY_USER}@${APP_DEPLOY_HOST}" "docker system prune -f -a ; docker image load"
```
Comment (ucyo) [inline .travis.yml:62]: What happens if the cleanup fails for some reason? Then the new image will not be loaded, right? 

And is there an edge case that the container is not running while this is being executed? If that happens and `image load` fails for some reason there are no images of anything on the machine

Comment (daniel-j-h) [reply .travis.yml:62]: > What happens if the cleanup fails for some reason? Then the new image will not be loaded, right?

Wrong :P That's exactly the reason we are doing `cmd1 ; cmd2` and not `cmd1 && cmd2`.

And yes in the case nothing is running _and_ image load fails for some reason (why?) then we're out of luck and e.g. have to manually deploy or make an (potentialy empty) commit to kick off another deploy.

Do you think there's a better way and a safer way to do this? :thinking:

Comment (ucyo) [reply .travis.yml:62]: No. Seems valid. Loading could fail if there is no connection to the machine. But it is a pretty rare case and an empty commit could fix it

**Pull request merged**